src/subclass/cell_renderer.rs

```diff
@@ -0,0 +1,12 @@
+use glib::subclass::prelude::*;
+use glib::ObjectClass;
+
+use CellRendererClass;
+
+pub trait CellRendererImpl: ObjectImpl + 'static {}
+
+unsafe impl<T: ObjectSubclass + CellRendererImpl> IsSubclassable<T> for CellRendererClass {
+    fn override_vfuncs(&mut self) {
+        <ObjectClass as IsSubclassable<T>>::override_vfuncs(self);
+    }
+}
```

Comment:
The `CellRenderer` class has various vfuncs. You don't need any of those for what you're doing?

Reply:
it's more like "cut and paste" I'm doing here. Likely I will need them.

Reply:
Ok, I guess you'll send a new PR at that point then :)

Reply:
or maybe do it in that one.... just need to figure out what to do.

Reply:
Ask on IRC if you need help :)

Reply:
Ok. It feels very complicated, but I think I got it, at least a draft....

Reply:
> Ok. It feels very complicated, but I think I got it, at least a draft....

Looks good to me mostly :) To give some background.

It's indeed a bit complicated and we can't unfortunately autogenerate this even in your case here because of the default handling when chaining up to the parent class' vfunc. That part we must get right as we always override all vfuncs, so even if you subclass does nothing then the chaining-up code is called and has to behave correctly.

Bindings for languages with a virtual machine/interpreter don't have that problem because they only override the vfuncs at runtime for the ones that are actually used by the subclass (e.g. Python). We can't do that in Rust because that information does not exist at runtime or even compile-time.

Other bindings just always do nothing and if there's a return value return "some default" and pray that all goes well (C++ for example), which we can't do because soundness is important for Rust developers. In C++ bindings can get away with blaming the developer if stuff crashes or misbehaves (you should've overridden this virtual method!) but in Rust that's not a choice for safe API if someone is supposed to take the bindings seriously.

I'm not sure how this can be improved. For the common cases we could probably get annotations (default return value / do nothing) but in other cases the default behaviour is complicated and nothing you really want to put into a textual annotation.